packages/serverpod/lib/src/server/endpoint_dispatch.dart

```diff
@@ -211,24 +328,19 @@ abstract class EndpointDispatch {
     return null;
   }
 
-  dynamic _formatArg(
-      dynamic input, Type type, SerializationManager serializationManager) {
-    return serializationManager.deserialize(input, type);
-  }
-
   /// Parses query parameters from a string into a map of parameters formatted
   /// according to the provided [ParameterDescription]s.
   ///
   /// Throws an exception if required parameters are missing or if the
   /// paramString can't be jsonDecoded.
+  @visibleForTesting
```

Comment:
Suggestion: If this method needs to be tested individually I would suggest moving to another file and use it from there.

I'm not a big fan of `visibleForTesting` tags. If we promise an external interface we might as well expose it (but not necessarily export it from the Serverpod package).

Reply:
Agree! Placing it in a helper file should be the best solution then 👍

Reply:
[f1ca25c](https://github.com/serverpod/serverpod/pull/2676/commits/f1ca25cf356181e0382a1b16027752640672b975)